### ISSUE TITLE
Variable length trajectories via padding in "summary_variables" batch for TimeSeriesTransformer

### ISSUE THREAD

**Issue by han-ol** [stefanradev93/BayesFlow#713]
**Is your feature request related to a problem? Please describe.**
Varying length time-series are unavoidable when sequence lengths are stochastic. `TimeSeriesTransformer` currently only supports masking of attention, and not yet masking for right-padded time-series.

**Describe the solution you'd like**
`TimeSeriesTransformer.call` has no mask argument at the moment, so `summary_mask` is unused. I think we should be able to deal with time series padding through a (B,T) shaped mask. 
We can create a attention mask from the (B,T), but also time embedding and pooling need to ensure invariance to adding pads.
The required changes are small, but break old serialized checkpoints.

**Describe alternatives you've considered**
1. Just passing summary_attention_mask created by hand:
* does not suffice, because pooling and time2vec inject dependence on number of pads (unnecessarily).
* requires users to understand attention internals unnecessarily
2. Always using `batch_size` of 1 sidesteps the need for padding time-series.
